sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RetryPolicy.java

```diff
@@ -20,25 +25,67 @@
  */
 public class RetryPolicy implements HttpPipelinePolicy {
 
-    private static final String RETRY_AFTER_MS_HEADER = "retry-after-ms";
-
     private final ClientLogger logger = new ClientLogger(RetryPolicy.class);
     private final RetryStrategy retryStrategy;
 
+    private final String retryAfterHeader;
+    private final ChronoUnit retryAfterTimeUnit;
+
     /**
-     * Creates a default {@link ExponentialBackoff} retry policy.
+     * Creates a default {@link ExponentialBackoff} retry policy. It will not use {@code retryAfterHeader}
+     * in {@link HttpResponse}.
      */
     public RetryPolicy() {
-        this(new ExponentialBackoff());
+        this(new ExponentialBackoff(), null, null);
+    }
+
+    /**
+     * Creates a default {@link ExponentialBackoff} retry policy along with provided {@code retryAfterHeader} and
+     * {@code retryAfterTimeUnit}.
+     *
+     * @param retryAfterHeader The 'retry-after' HTTP header name to lookup for the retry duration.The value
+     * {@code null} is valid.
+     * @param retryAfterTimeUnit The time unit to use while applying retry based on value specified in
+     * {@code retryAfterHeader} in {@link HttpResponse}.The value {@code null} is valid only in case when
+     * {@code retryAfterHeader} is empty or {@code null}.
+     * @throws NullPointerException Only if {@code retryAfterTimeUnit} is {@code null} and {@code retryAfterHeader}
+     * is not {@code null}.
+     */
+    public RetryPolicy(String retryAfterHeader, ChronoUnit retryAfterTimeUnit) {
```

Comment:
As @srnagar, you, and I discussed let's use a properties class to pass these values instead, so we'll have three constructors for RetryPolicy

1. `RetryPolicy()` defaults to exponential back off which may or may not respect retry after headers
2. `RetryPolicy(RetryStrategy)`
3. `RetryPolicy(RetryOptions)`

`RetryOptions` will contain the options to configure how retrying will occur such as the `RetryStrategy` which will default to ExponentialBackoff and headers to look for to determine retry delay which may or may not be ignored by default.

Reply:
Since we have identified 4-5 more Policies which will need `Options` , I would generalize that all the policy have two constructor similar to following
1. `RetryPolicy()`  
2. `RetryPolicy(RetryOptions)`  

And Options is where we will configure strategy, retry, header  etc. All the variations.

Reply:
adding old constructor back `RetryPolicy(RetryStrategy)`